CumulusCI v3.35.0 (#2619)
SFDO-Tooling/CumulusCI: HISTORY.rst

History
=======

3.35.0 (2021-05-13)
-------------------

Critical Changes

> - Upgraded robot framework to 4.x. For information about new features and some backward incompatibilities see the `Robot Framework 4.0 release notes <https://github.com/robotframework/robotframework/blob/master/doc/releasenotes/rf-4.0.rst>`_.

Reviewer: ```suggestion
- Upgraded Robot Framework to 4.x. For information about new features and some backward incompatibilities see the `Robot Framework 4.0 release notes <https://github.com/robotframework/robotframework/blob/master/doc/releasenotes/rf-4.0.rst>`_.
```

> - ``update_dependencies`` now guarantees to resolve unpackaged metadata directories (subdirectories of ``unpackaged/pre`` and ``unpackaged/post``) in alphabetical order, matching the behavior of ``deploy_pre`` and ``deploy_post``. ``unpackaged/pre/bar`` will deploy prior to ``unpackaged/pre/foo``. The previous behavior was undefined, which caused rare problems. This change is critical only for projects that have deployment-order dependencies between unpackaged directories located in upstream dependencies and rely on the current undefined load order.

Reviewer: ```suggestion
- The ``update_dependencies`` task now guarantees to resolve unpackaged metadata directories (subdirectories of ``unpackaged/pre`` and ``unpackaged/post``) in alphabetical order, matching the behavior of ``deploy_pre`` and ``deploy_post``. ``unpackaged/pre/bar`` will deploy prior to ``unpackaged/pre/foo``. The previous behavior was undefined, which caused rare problems. This change is critical only for projects that have deployment-order dependencies between unpackaged directories located in upstream dependencies and rely on the current undefined load order.
```

Changes

> - The CumulusCI documentation has a new section: `Testing with Second-Generation Packaging <https://cumulusci.readthedocs.io/en/latest/2gp_testing.html>`_

Reviewer: I like calling this out!

> - CumulusCI has two new service types: ``oauth2_client`` & ``marketing_cloud``

Reviewer: ```suggestion
- CumulusCI has two new service types: ``oauth2_client`` & ``marketing_cloud``. These are considered experimental.
```

> - The ``marketing_cloud`` service allows users to connect to a marketing cloud tenant via OAuth so that future tasks that work with Marketing cloud can make API calls on the user's behalf.

Reviewer: ```suggestion
- The ``marketing_cloud`` service allows users to connect to a Marketing Cloud tenant via OAuth so that tasks that work with Marketing Cloud can make API calls on the user's behalf.
```

> - The ``oauth2_client`` service takes information for an individual OAuth2 client which can then be used in place of the default client. This currently applies only to the marketing-cloud service. To setup a marketing cloud service with a custom oauth-client use: ``cci service connect marketing-cloud <name-of-service> --oauth_client <name-of-oauth-client>``.

Reviewer: ```suggestion
- The ``oauth2_client`` service takes information for an individual OAuth2 client which can then be used in place of the default client. This currently applies only to the ``marketing_cloud`` service. To setup a Marketing Cloud service with a specific OAuth2 client use: ``cci service connect marketing-cloud <name-of-service> --oauth_client <name-of-oauth-client>``.
```

> - CumulusCI has a new task: ``deploy_marketing_cloud_package``. This task allows a user to pass the path to a .zip file to a marketing cloud package (downloaded from the Marketing Cloud package manager) and deploy the package via a ``marketing_cloud`` service (see above).

Reviewer: ```suggestion
- CumulusCI has a new task: ``deploy_marketing_cloud_package``. This task allows a user to pass the path to a .zip file to a Marketing Cloud package (downloaded from the Marketing Cloud Package Manager) and deploy the package via a ``marketing_cloud`` service (see above). Note that successfully deploying a package using this task may require permissions that are not generally available.
```

> - The ``DeployBundles`` task avoids warnings by freezing installer steps that match current expectations.

Reviewer: ```suggestion
- The ``deploy_pre`` and ``deploy_post`` tasks avoid warnings by freezing installer steps that match current expectations.
```

and I would move this under Issues Closed
